Marlin/ultralcd.cpp

```diff
@@ -1083,6 +1092,51 @@ void lcd_cooldown() {
 
 /**
  *
+ * "Change Filament" submenu
+ *
+ */
+static void lcd_change_filament_menu() {
+  START_MENU();
+  MENU_ITEM(back, MSG_MAIN);
+  MENU_ITEM(function, MSG_REMOVE_FILAMENT , lcd_retract_filament);
+  MENU_ITEM(submenu, MSG_INSERT_FILAMENT , lcd_insert_filament_menu);
+  END_MENU();
+}
+
+static void lcd_insert_filament_menu() {
+  START_MENU();
+  MENU_ITEM(back, MSG_CHANGE_FILAMENT);
+  MENU_ITEM_EDIT_CALLBACK(bool, MSG_INSERT_FILAMENT_SLOW ,&filamentSlowInsert ,lcd_insert_filament_slow);
+  if(!filamentSlowInsert)
+    MENU_ITEM(function, MSG_INSERT_FILAMENT  , lcd_insert_filament);
+  END_MENU();
+}
+
+static void lcd_insert_filament_slow(){
+ while (filamentSlowInsert) {
```

Comment:
This `while` loop will work because `st_synchronize` keeps the lcd active. But I'm not sure if it's the best approach here. I will have to try it out and see how it behaves.